ChattoAdditions/Source/Input/ExpandableTextView.swift

```diff
@@ -225,4 +240,26 @@ open class ExpandableTextView: UITextView {
         self.placeholder.textContainerInset = self.textContainerInset
         self.placeholder.backgroundColor = UIColor.clear
     }
+
+    // When you press on inset area inside UITextView, cursor is automatically moved to beginning of the content.
+    // We move press point to the closest point inside text container to avoid this behaviour.
+    // Point that is already inside text container or outside of the view itself will not be moved.
+    private func closestPointInTextContainer(to point: CGPoint) -> CGPoint {
+        guard self.bounds.contains(point) else { return point }
+        let x = min(
+            max(
+                point.x,
+                self.textContainerInset.left
+            ),
+            self.bounds.width - self.textContainerInset.right
+        )
+        let y = min(
+            max(
+                point.y,
+                self.textContainerInset.top
+            ),
+            self.bounds.height - self.textContainerInset.bottom
+        )
+        return CGPoint(x: x, y: y)
+    }
```

Comment:
I recommend to introduce a "clamp" function:
```
public extension Comparable {
    func clamp(_ minVal: Self, _ maxVal: Self) -> Self {
        return Swift.max(minVal, Swift.min(maxVal, self))
    }
}
```

Then you can rewrite this method this way:
```
private func closestPointInTextContainer(to point: CGPoint) -> CGPoint {
    guard self.bounds.contains(point) else { return point }
    let x = point.x.clamp(self.textContainerInset.left, self.bounds.width - self.textContainerInset.right)
    let y = point.y.clamp(self.textContainerInset.top, self.bounds.height - self.textContainerInset.bottom)
    return CGPoint(x: x, y: y)
}
```

Reply:
Thanks, will do